microsoft-edge/progressive-web-apps-chromium/how-to/service-workers.md

```diff
@@ -41,63 +73,282 @@ The lifecycle of a Service Worker consists of multiple steps, with each step tri
+
+```javascript
+// The name of the new cache your app uses.
+const CACHE_NAME = "my-app-cache-v2";
+// The list of static files your app needs to start.
+const PRE_CACHED_RESOURCES = ["/", "styles-124656.css", "app-576391.js"];
+
+// Listen to the `install` event.
+self.addEventListener("install", event => {
+  async function preCacheResources() {
+    // Open the app's cache.
+    const cache = await caches.open(CACHE_NAME);
+    // Cache the new static resources.
+    cache.addAll(PRE_CACHED_RESOURCES);
+  }
+
+  event.waitUntil(preCacheResources());
+});
+
+// Listen to the `activate` event to clear old caches.
+self.addEventListener("activate", event => {
+  async function deleteOldCaches() {
+    // List all caches by their names.
+    const names = await caches.keys();
+    await Promise.all(names.map(name => {
+      if (name !== CACHE_NAME) {
+        // If a cache's name is the current name, delete it.
+        return caches.delete(name);
+      }
+    }));
+  }
+
+  event.waitUntil(deleteOldCaches());
+});
+```
+
+Compare the `CACHE_NAME` and `PRE_CACHED_RESOURCES` values between the above code snippet and the one in [Pre-cache resources](#pre-cache-resources). When this new Service Worker is installed, a new cache will be created and the new static resources will be downloaded and cached. When the Service Worker is activated, the old cache will be deleted. At this point, the user will have the new version of the app.
+
+Making changes to your Service Worker can sometimes be complex. Use a library such as [Workbox](https://developer.chrome.com/docs/workbox/) to simplify your static resources build step and Service Worker code.
+
+
+<!-- ====================================================================== -->
+## Test for network connections in your PWA
+
+It's helpful to know when a network connection is available, in order to synchronize data or inform users that the network status has changed.
+
+Use the following options to test for network connectivity:
+
+#### navigator.onLine
+
+The `navigator.onLine` property is a boolean that lets you know the current status of the network. If the value is `true`, the user is online; otherwise, the user is offline.
+
+To learn more, see [navigator.onLine](https://developer.mozilla.org/docs/Web/API/NavigatorOnLine) on MDN.
+
+#### Online and Offline Events
+
+You can take action when your network connectivity changes.  You can listen and take action in response to network events.  The events are available on the `window`, `document`, and `document.body` elements, as shown below:
+
+```javascript
+window.addEventListener("online",  function(){
+    console.log("You are online!");
+});
+window.addEventListener("offline", function(){
+    console.log("Oh no, you lost your network connection.");
+});
+```
+
+To learn more, see [Online and Offline Events](https://developer.mozilla.org/docs/Web/API/NavigatorOnLine/Online_and_offline_events) on MDN.
```

Comment:
link isn't working, to confirm link text.  repro: 
go to:
https://developer.mozilla.org/docs/Web/API/NavigatorOnLine/Online_and_offline_events
get redir'd to:
https://developer.mozilla.org/en-US/docs/Web/API/Navigator/onLine - "Navigator.onLine", not "Online and Offline Events".